server/resources/resource_management.go

```diff
@@ -78,31 +78,40 @@ func measureRequestTime(timeStarted time.Time, requestType string, schemaId stri
 	metrics.UpdateTimer(timeStarted, "req.%s.%s", schemaId, requestType)
 }
 
-//InTransaction executes function in the db transaction and set it to the context
-func InTransaction(ctx middleware.Context, dataStore db.DB, level transaction.Type, f func() error) error {
+//resourceTransactionWithContext executes function in the db transaction and set it to the context
+func resourceTransactionWithContext(ctx middleware.Context, dataStore db.DB, level transaction.Type, fn func() error) error {
 	if ctx["transaction"] != nil {
 		return fmt.Errorf("cannot create nested transaction")
 	}
 
-	aTransaction, err := dataStore.BeginTx(context.Background(), &transaction.TxOptions{IsolationLevel: level})
-	if err != nil {
-		return fmt.Errorf("cannot create transaction: %v", err)
+	org := middleware.Context{}
```

Comment:
Too short parameter name so I need time to understand what this is. You should change originalCtx and need proper comment why this is necessary.